### PR TITLE
Home Task week 1 - OOD and OOP

### PR DESCRIPTION
**Pull request by rajeshmoka22**

### CODE REVIEW

File: OOD and OOP/src/Shape.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        if(points.length >2) this.points = points;
+        else throw new Error('shape should atleast have 3 points');
+    }
+
+    toString(): string {
+        let pointsStringArr: Array<String> = [];
+        this.points.forEach(point => {
+            pointsStringArr.push(` (${point.x}, ${point.y})`);
+        });
+        return `A Shape with color of ${this.color} and ${this.filled? 'filled' : 'not filled'}. Points:${pointsStringArr.join(',')}.`
```
Comment (SomeshThakur) [inline OOD and OOP/src/Shape.ts:23]: .join() is optional as the template string will convert the array to string with `,` as a delimiter.

File: OOD and OOP/src/Shape.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    getPerimeter(): number {
+        let perimeter = 0;
+        const pointsLen = this.points.length;
+        if(pointsLen >=3) {
```
Comment (SomeshThakur) [inline OOD and OOP/src/Shape.ts:29]: Condition is not required as we are not allowing the shape to be created without having 3 points at least.

File: OOD and OOP/src/Shape.ts (hunk 1 above)
Comment (rajeshmoka22) [reply OOD and OOP/src/Shape.ts:23]: Yeah got it.

File: OOD and OOP/src/Shape.ts (hunk 2 above)
Comment (rajeshmoka22) [reply OOD and OOP/src/Shape.ts:29]: Got used to checking the data before performing operation on it. That's a good point.

File: OOD and OOP/src/Triangle.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Shape } from './Shape';
+import { Point } from './Point';
+
+export class Triangle extends Shape {
+
+    constructor(point1: Point, point2: Point, point3: Point)
+    constructor(point1: Point, point2: Point, point3: Point, color: string, filled: boolean)
+    constructor(point1: Point, point2: Point, point3: Point, color?: string, filled?: boolean) {
+        if(color && filled) {
```
[the remark this answers is not in the document]
Comment (rajeshmoka22) [reply OOD and OOP/src/Triangle.ts:9]: Yeah got it. making changes now.

File: OOD and OOP/src/Triangle.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        } else triangleType = 'scalene triangle';
+        return triangleType;
+    }
+    
+    toString(): string {
+        let pointsArr: Array<String> = [];
+        this.points.forEach((point, index) => {
+            pointsArr.push(`v${index+1}=(${point.x}, ${point.y})`);
+        });
+        return `Triangle[${pointsArr.join(',')}]`;
```
[the remark this answers is not in the document]
Comment (rajeshmoka22) [reply OOD and OOP/src/Triangle.ts]: Yeah updated it.

File: OOD and OOP/src/Shape.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            for(let i=0; i<pointsLen-1; i++) {
```
[the remark this answers is not in the document]
Comment (rajeshmoka22) [reply OOD and OOP/src/Shape.ts]: I went for traditional for loop because I am not iterating through entire array.

**Pull request merged**